.github/actions/build-test/ubuntu-i386/action.yml

```diff
@@ -0,0 +1,115 @@
+name: Build & Test - Ubuntu - i386
+
+inputs:
+  GH_TOKEN:
+    required: true
+  BOOST_VERSION:
+    required: true
+  THRIFT_VERSION:
+    required: true
+  WARN_AS_ERR:
+    required: true
+  BUILD_TYPE:
+    required: true
+  SHARED_LIBS_TOGGLE:
+    required: true
+  OPENSSL_TOGGLE:
+    required: true
+  RUN_TESTS:
+    required: true
+  HAZELCAST_ENTERPRISE_KEY:
+    required: true
+  AWS_ACCESS_KEY_ID:
+    required: true
+  AWS_SECRET_ACCESS_KEY:
+    required: true
+  HZ_TEST_AWS_INSTANCE_PRIVATE_IP:
+    required: true
+
+env:
+  # Not possible to set this as a default
+  # https://github.com/orgs/community/discussions/46670
+  shell: bash
+
+runs:
+  using: composite
+  steps:
+    - name: Read Config
+      shell: ${{ env.shell }}
+      run: cat .github/config.env >> $GITHUB_ENV
+
+    # Install Java via `apt`, can't use the `setup-java` action on this image
+    # `setup-java` is written in TypeScript and requires a compatible Node installation
+    # Newest available version for this image is Node 8, which is too old for any version to run against
+    # https://github.com/actions/setup-node/issues/922
+    - name: Install Necessary Packages
+      shell: ${{ env.shell }}
+      run: |
+        apt-get update
+        apt-get install -y build-essential cmake curl git libssl-dev maven net-tools openjdk-${{ env.JAVA_VERSION }}-jre-headless gdb curl
+
+    - name: Make sure the target architecture is 32 bit
+      shell: ${{ env.shell }}
+      run: |
+        echo 'int main() { return sizeof(void*) != 4; }' > test.c
+        gcc test.c -oa
+        ./a
+        rm a test.c
+
+    # `apt-get` brings in `3.6` which is too old to be compatible with Java 17
+    - name: Install Maven
+      shell: ${{ env.shell }}
+      run: |
+        install_dir="/opt/maven"
+        mkdir ${install_dir}
```

Comment:
move this to .sh like boost

Reply:
The boost installer is used in more than one place, so sharing makes sense - but this isn't, so doesn't, imo.

Reply:
ok. I would have prefer one or the other approach. 
In the future might need it somewhere else